drivers/adc/ad4170/ad4170.c

```diff
@@ -0,0 +1,2119 @@
+ */
+int ad4170_read16s(struct ad4170_dev *dev, uint32_t *pbuf,
+		   uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_16b_STATUS);
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_24
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4170_read24(struct ad4170_dev *dev, uint32_t *pbuf,
+		  uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_24b);
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_24_Status
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4170_read24s(struct ad4170_dev *dev, uint32_t *pbuf,
+		   uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_24b_STATUS);
+}
+
+/**
+ * @brief Read multiple samples using direct register access from Data_32b
+ * @param dev - The device structure.
+ * @param pbuf - Output pointer where to store the data values.
+ * @param nb_samples - The number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4170_read32(struct ad4170_dev *dev, uint32_t *pbuf,
+		  uint16_t nb_samples)
+{
+	return _ad4170_read(dev, pbuf, nb_samples, AD4170_REG_DATA_32b);
+}
+
+static int32_t _scratchpad_test(struct ad4170_dev *dev, uint8_t pattern)
+{
+	int ret;
+	uint32_t reg;
+
+	ret = ad4170_spi_reg_write(dev, AD4170_REG_SCRATCH_PAD, pattern);
+	if (ret)
+		return ret;
+
+	ret = ad4170_spi_reg_read(dev, AD4170_REG_SCRATCH_PAD, &reg);
+	if (ret)
+		return ret;
+	if (reg != pattern)
+		ret = -EFAULT;
+
+	return 0;
+}
+
+static int32_t _ad4170_continuous_read_one(struct ad4170_dev *dev,
+		uint32_t *data_out, uint8_t *status_out, bool exit)
+{
+	int ret;
+	uint8_t rdyb, icrc;
+	uint32_t timeout;
+	uint8_t buf[5];
+	uint16_t sz = 3;
+	bool status_en = dev->config.adc_ctrl.cont_read_status_en;
+	sz += (uint16_t)status_en;
+	sz += (uint16_t)dev->spi_settings.crc_enabled;
+
+	if (!status_out && status_en)
+		return -EINVAL;
+
+	// this function relies on dig_aux1 pin being configured as RDY signal
+	if (!dev->gpio_dig_aux1
+	    || (dev->config.pin_muxing.dig_aux1_ctrl != AD4170_DIG_AUX1_RDY))
+		return -ENOTSUP;
+
+	memset(buf, 0, sz);
+	if (exit)
+		buf[0] = 0xa5;
+
+	rdyb = NO_OS_GPIO_HIGH;
+	timeout = dev->rdy_conv_timeout;
+	while (rdyb == NO_OS_GPIO_HIGH && timeout--) {
+		no_os_gpio_get_value(dev->gpio_dig_aux1, &rdyb);
+	};
+	if (!timeout)
+		return -ETIMEDOUT;
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buf, sz);
+	if (ret)
+		return ret;
+
+	// TODO: handle 32b width. By default 24b data width is selected.
```

Comment:
Should we remove the TODOs?

Reply:
I'm not sure about this one, looks like there is some missing functionality but I can't add it without hardware setup, so I'd rather leave it in for now.

Reply:
actually i re-read the datasheet and looks like continuous read works with 24b mode only so I guess we're good, I'm removing the TODO and the 32b handling that was commented out